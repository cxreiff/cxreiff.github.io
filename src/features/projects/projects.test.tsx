import { render, fireEvent } from '@testing-library/preact'

import Projects from './projects.component'

const mockNavigate = jest.fn()
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockNavigate,
}))

test('should render', () => {
    const { container } = render(<Projects />)

    expect(container).toMatchSnapshot()
})

test('should navigate to each project', () => {
    const { getByText } = render(<Projects />)

    fireEvent.click(getByText('asteroids game'))
    expect(mockNavigate).toHaveBeenCalledWith('/projects/pixijs/asteroids')

    fireEvent.click(getByText('babylon.js project'))
    expect(mockNavigate).toHaveBeenCalledWith('/projects/babylonjs')
})
