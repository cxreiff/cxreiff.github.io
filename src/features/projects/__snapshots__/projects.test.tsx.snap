// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projects should render 1`] = `
<Wrapper>
  <Provider
    store={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Provider
      value={
        Object {
          "store": Object {
            "@@observable": [Function],
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "subscription": Object {
            "addNestedSub": [Function],
            "getListeners": [Function],
            "handleChangeWrapper": [Function],
            "isSubscribed": [Function],
            "notifyNestedSubs": [Function],
            "onStateChange": [Function],
            "trySubscribe": [Function],
            "tryUnsubscribe": [Function],
          },
        }
      }
    >
      <HashRouter>
        <Router
          location={
            Object {
              "hash": "",
              "key": "default",
              "pathname": "/",
              "search": "",
              "state": null,
            }
          }
          navigationType="POP"
          navigator={
            Object {
              "action": "POP",
              "back": [Function],
              "block": [Function],
              "createHref": [Function],
              "forward": [Function],
              "go": [Function],
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "default",
                "pathname": "/",
                "search": "",
                "state": null,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Provider
            value={
              Object {
                "basename": "/",
                "navigator": Object {
                  "action": "POP",
                  "back": [Function],
                  "block": [Function],
                  "createHref": [Function],
                  "forward": [Function],
                  "go": [Function],
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "key": "default",
                    "pathname": "/",
                    "search": "",
                    "state": null,
                  },
                  "push": [Function],
                  "replace": [Function],
                },
                "static": false,
              }
            }
          >
            <Provider
              value={
                Object {
                  "location": Object {
                    "hash": "",
                    "key": "default",
                    "pathname": "/",
                    "search": "",
                    "state": null,
                  },
                  "navigationType": "POP",
                }
              }
            >
              <Projects>
                <section
                  className="projects"
                >
                  <Card
                    image=""
                    onClick={[Function]}
                    primary="pixi.js project"
                    secondary="pixi.js"
                  >
                    <article
                      className="card card_clickable"
                      onClick={[Function]}
                    >
                      <header>
                        <h4>
                          pixi.js project
                        </h4>
                      </header>
                      <footer>
                        pixi.js
                      </footer>
                    </article>
                  </Card>
                  <Card
                    image=""
                    onClick={[Function]}
                    primary="babylon.js project"
                    secondary="babylon.js"
                  >
                    <article
                      className="card card_clickable"
                      onClick={[Function]}
                    >
                      <header>
                        <h4>
                          babylon.js project
                        </h4>
                      </header>
                      <footer>
                        babylon.js
                      </footer>
                    </article>
                  </Card>
                </section>
              </Projects>
            </Provider>
          </Provider>
        </Router>
      </HashRouter>
    </Provider>
  </Provider>
</Wrapper>
`;
